SetupBLT.cmake

```diff
@@ -193,8 +193,10 @@ if (NOT BLT_LOADED)
     # File extension lists used to filter sources based on languages for code checks
     # and filtering Fortran sources out of cuda/hip source lists
     # Note: this filtering is case-insensitive
-    set(BLT_C_FILE_EXTS ".cpp" ".hpp" ".cxx" ".hxx" ".c" ".h" ".cc" ".hh" ".inl" ".cu" ".cuh"
-               CACHE STRING "List of known file extensions used for C/CXX sources")
+    set(BLT_C_FILE_EXTS ".c" ".h"
+               CACHE STRING "List of known file extensions used for C sources")
+    set(BLT_CXX_FILE_EXTS ".cpp" ".hpp" ".cxx" ".hxx" ".c" ".h" ".cc" ".hh" ".inl" ".cu" ".cuh"
```

Comment:
Is it a problem if .c and .h are in both BLT_C_FILE_EXTS and BLT_CXX_FILE_EXTS?

Reply:
It's looking like .c and .h will always be added to C_LIST regardless of whether CXX_LIST is defined.

Reply:
Just for my own understanding: It shouldn't matter whether .h is in C_LIST or CXX_LIST, right? Or does the source language property matter for header files?

Reply:
Just a copy-past error. The macro is backwards compatible, and the only difference is that the new CXX_LIST should _only_ have C++ files in it since HIP is C++. CMake docs says: "Setting this property for a file means this file will be compiled, unless HEADER_FILE_ONLY is set."